File: screenshot/Views/CollectionViewCells/ShoppableCollectionViewCell.m
//
//  ShoppableCollectionViewCell.m
//  screenshot
//
//  Created by Corey Werner on 8/21/17.
//  Copyright © 2017 crazeapp. All rights reserved.
//

#import "ShoppableCollectionViewCell.h"
#import "screenshot-Swift.h"

@interface ShoppableCollectionViewCell ()

@property (nonatomic, strong) UIImageView *imageView;

@end

@implementation ShoppableCollectionViewCell

- (instancetype)initWithFrame:(CGRect)frame {
    self = [super initWithFrame:frame];
    if (self) {
        self.contentView.backgroundColor = [UIColor whiteColor];
        self.layer.borderColor = [self borderColor].CGColor;
        self.layer.borderWidth = 1.f;
        
        _imageView = ({
            UIImageView *imageView = [[UIImageView alloc] init];
            imageView.translatesAutoresizingMaskIntoConstraints = NO;
            imageView.contentMode = UIViewContentModeScaleAspectFit;
            [self.contentView addSubview:imageView];
            [imageView.topAnchor constraintEqualToAnchor:self.contentView.topAnchor].active = YES;
            [imageView.leadingAnchor constraintEqualToAnchor:self.contentView.leadingAnchor].active = YES;
            [imageView.bottomAnchor constraintEqualToAnchor:self.contentView.bottomAnchor].active = YES;
            [imageView.trailingAnchor constraintEqualToAnchor:self.contentView.trailingAnchor].active = YES;
            imageView;
        });
    }
    return self;
}

- (void)setSelected:(BOOL)selected {
    [super setSelected:selected];
    
    self.layer.borderColor = selected ? [UIColor crazeRed].CGColor : [self borderColor].CGColor;
}

- (UIColor *)borderColor {
    return [UIColor colorWithWhite:193.f/255.f alpha:1.f];
}

- (void)setImage:(UIImage *)image {
    self.imageView.image = image;
}

- (UIImage *)image {
    return self.imageView.image;
}

@end

File: screenshot/View/Shoppable/ShoppableCollectionViewCell.swift
//
//  ShoppableCollectionViewCell.swift
//  screenshot
//
//  Created by Corey Werner on 11/15/17.
//  Copyright © 2017 crazeapp. All rights reserved.
//

import Foundation
import UIKit

class ShoppableCollectionViewCell : UICollectionViewCell {
    private let imageView = UIImageView()
    private let borderColor = UIColor.gray8
    
    var image: UIImage? {
        set {
            imageView.sd_cancelCurrentImageLoad()
            imageView.image = newValue
        }
        get {
            return imageView.image
        }
    }
    var imageUrl:String? {
        didSet {
            imageView.sd_cancelCurrentImageLoad()
            imageView.image = nil
            if let imageUrl = imageUrl, let url = URL(string:imageUrl) {
                imageView.sd_setImage(with: url, completed: nil)
            }
        }
    }
    
    required init?(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)
    }
    
    override init(frame: CGRect) {
        super.init(frame: frame)
        
        contentView.backgroundColor = .white
        layer.borderColor = borderColor.cgColor
        layer.borderWidth = 1.5
        layer.cornerRadius = .defaultCornerRadius
        layer.masksToBounds = true
        
        imageView.translatesAutoresizingMaskIntoConstraints = false
        imageView.contentMode = .scaleAspectFit
        contentView.addSubview(imageView)
        imageView.topAnchor.constraint(equalTo: contentView.topAnchor).isActive = true
        imageView.leadingAnchor.constraint(equalTo: contentView.leadingAnchor).isActive = true
        imageView.bottomAnchor.constraint(equalTo: contentView.bottomAnchor).isActive = true
        imageView.trailingAnchor.constraint(equalTo: contentView.trailingAnchor).isActive = true
    }
    
    override var isSelected: Bool {
        didSet {
            layer.borderColor = isSelected ? UIColor.crazeRed.cgColor : borderColor.cgColor
        }
    }
}
